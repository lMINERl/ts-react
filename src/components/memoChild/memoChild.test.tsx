import * as React from 'react';
import { shallow } from '../../test/Configurations';
import MemoChild from './memoChild';
import { render, getByTestId } from '@testing-library/react';

describe('<MemoChild />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<MemoChild />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has number display', () => {
    const { container } = render(<MemoChild />);
    const renderCount = getByTestId(container, 'render-count');
    expect(renderCount.textContent).toMatch(/[a-z A-Z]* \: [0-9]+/);
  });

  it('should increase render-count on rerender', () => {
    const { container, rerender } = render(<MemoChild />);
    const renderCount = getByTestId(container, 'render-count');
    expect(renderCount.textContent).toMatch(/[a-z A-Z]* \: 1/);
    rerender(<MemoChild />);
    expect(renderCount.textContent).toMatch(/[a-z A-Z]* \: 2/);
  });
});
