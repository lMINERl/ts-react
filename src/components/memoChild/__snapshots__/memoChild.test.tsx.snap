// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoChild /> renders without crashing 1`] = `ShallowWrapper {}`;
